src/cli/update.rs

```diff
@@ -0,0 +1,386 @@
+                    }
+                }
+                environment_diff.insert(platform, diff);
+            }
+
+            // Remove empty diffs
+            environment_diff.retain(|_, diff| !diff.is_empty());
+
+            result
+                .environment
+                .insert(environment_name.to_string(), environment_diff);
+        }
+
+        // Find environments that were completely removed
+        for (environment_name, environment) in previous
+            .environments()
+            .filter(|(name, _)| !result.environment.contains_key(*name))
+            .collect_vec()
+        {
+            let mut environment_diff = IndexMap::new();
+            for (platform, packages) in environment.packages_by_platform() {
+                let mut diff = PackagesDiff::default();
+                for package in packages {
+                    match package {
+                        Package::Conda(p) => {
+                            diff.removed.push(Package::Conda(p));
+                        }
+                        Package::Pypi(p) => {
+                            diff.removed.push(Package::Pypi(p));
+                        }
+                    }
+                }
+                environment_diff.insert(platform, diff);
+            }
+            result
+                .environment
+                .insert(environment_name.to_string(), environment_diff);
+        }
+
+        // Remove empty environments
+        result.environment.retain(|_, diff| !diff.is_empty());
+
+        result
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.environment.is_empty()
+    }
+
+    // Format the lock-file
+    pub fn print(&self) {
+        enum Change<'i> {
+            Added(&'i Package),
+            Removed(&'i Package),
+            Changed(&'i Package, &'i Package),
+        }
+
+        fn format_package_identifier(package: &Package) -> String {
+            match package {
+                Package::Conda(p) => format!(
+                    "{} {}",
+                    &p.package_record().version.as_str(),
+                    &p.package_record().build
+                ),
+                Package::Pypi(p) => p.data().package.version.to_string(),
+            }
+        }
+
+        for (environment_name, environment) in
+            self.environment.iter().sorted_by(|(a, _), (b, _)| a.cmp(b))
+        {
+            println!(
+                "Environment: {}",
+                consts::ENVIRONMENT_STYLE.apply_to(environment_name),
+            );
+            for (platform, packages) in environment {
+                println!("  Platform: {}", consts::PLATFORM_STYLE.apply_to(platform));
+                itertools::chain!(
+                    packages.added.iter().map(Change::Added),
+                    packages.removed.iter().map(Change::Removed),
+                    packages.changed.iter().map(|a| Change::Changed(&a.0, &a.1))
+                )
+                .sorted_by_key(|c| match c {
```

Comment:
I think it would be awesome to improve the output in 2 ways. 

Personally I'm most interested by the `major` then `minor` then `patch` changes.
The added and removed packages are also a seperate thing.

So what if we sort on name, then on `Change` and then make the changed numbers in the version bold or something like that.

e.g. 
<pre>
Platform: <span style="color:#A2734C">linux-64</span>
    ~ annotated-types 0.<b>6.0</b> pyhd8ed1ab_0 -&gt; 0.<b>7.0</b> pyhd8ed1ab_0
    ~ libgcc-ng 13.2.0 h77fa898_<b>6</b> -&gt; 13.2.0 h77fa898_<b>7</b>
    - importlib-metadata 7.1.0 pyha770c72_0
    + importlib_resources 6.4.0 pyhd8ed1ab_0
</pre>